### DISCUSSION TITLE
Dynamically Change Orientation?

### DISCUSSION THREAD

**Discussion by lbussy** [Bodmer/TFT_eSPI#931] [Q&A - General] [answered]
In our application, I would like to be able to dynamically change the orientation.  Currently we are initializing like so:

``` cpp
    tft = new TFT_eSPI();
    tft->begin();
    tft->setFreeFont(&FreeSans12pt7b);
    tft->setSwapBytes(true);
    tft->initDMA();
    if (config.invertTFT)
    {
        tft->setRotation(1);
        delay(20);
    }
    else
    {
        tft->setRotation(3);
        delay(20);
    }
```

There's no `TFT_eSPI::end();` or analogue that I can find.  I did try:

``` cpp
    tft->deInitDMA();
    tft->endWrite();
    delete tft;
```

But, while doing a really good job of stopping the display, when I re-init as above the screen does not initialize.  I thought maybe there was something simple I was missing and you might be able to point me in the right direction?

**Comment by lbussy** [comment]
Hi @Bodmer, I was wondering if you had any ideas for me here?  Possible and I've missed something easy?

**Comment by Bodmer** [comment, marked as answer]
The library class for the TFT has no destructor and thus can not be "ended". The rationale was that if a TFT is connected then the instance can be global as it is unlikely the dislay will only be used once.  The memory footprint is quite low compared to the RTOS etc on an ESP32 so I have never seen the need to end the library instance.

You can change the rotation at any time, ([example](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/320%20x%20240/TFT_graphicstest_one_lib/TFT_graphicstest_one_lib.ino#L102-L108)) you do not need to stop and start with a new instance. The display can be re-initialised by calling tft.begin() again as there is a [boot lock](https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_eSPI.cpp#L514-L557) that only runs part of the initiailisation the second time it is called. This permits the display to be powered down and restarted later.

**Comment by lbussy** [comment]
Brilliant!  Just what I needed, thank you.  I'm glad I included the "why" when I asked.
